File: TheRayCode/Java/Creational/Builder/Example3/src/ConcreteBuilder.java
public class ConcreteBuilder implements Builder{

    private Product product = new Product();

    @Override
    public void buildPartA() {
        product.setPartA("Concrete Part A");
    }

    @Override
    public void buildPartB() {
        product.setPartB("Concrete Part B");
    }

    @Override
    public void buildPartC() {
        product.setPartC("Concrete Part C");
    }

    @Override
    public Product getResult() {
        return product;
    }
}

File: TheRayCode/CSharp/Creational/Builder/Example/Example/ConcreteBuilder.cs
namespace Example;

public class ConcreteBuilder: Builder
{
    private Product _product = new Product(); // The product instance being built

    public override void BuildPartA()
    {
        _product.Add("PartA"); // Adds PartA to the product
    }

    public override void BuildPartB()
    {
        _product.Add("PartB"); // Adds PartB to the product
    }

    public override Product GetResult()
    {
        return _product; // Returns the fully built product
    }
}